src/cargo/ops/cargo_new.rs

```diff
@@ -85,6 +95,58 @@ struct CargoNewConfig {
     version_control: Option<VersionControl>,
 }
 
+fn get_input_template(config: &Config, opts: &MkOptions) -> CargoResult<TemplateSet> {
+    let path = opts.path;
+    let name = opts.name;
+    let templates_dir = config.template_path();
+    if fs::metadata(&templates_dir).is_err() {
+        try!(fs::create_dir_all(&templates_dir));
+    }
+
+    let template_type = try!(get_template_type(opts.template_repo, opts.template));
+    let template_set = match template_type {
+        // given template is a remote git repository & needs to be cloned
+        TemplateType::GitRepo(repo_url) => {
+            let template_dir = try!(TempDir::new(name));
+            try!(config.shell().status("Cloning", repo_url));
+            try!(Repository::clone(repo_url, &*template_dir.path()).chain_error(|| {
```

Comment:
I guess the way forward here is to change `GitCheckout::clone_repo` and change source from a Path to a String so it can handle remote urls?